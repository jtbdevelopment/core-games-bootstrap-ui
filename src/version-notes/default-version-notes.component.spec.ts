import {TestBed} from '@angular/core/testing';
import {DefaultVersionNotesComponent} from './default-version-notes.component';
import {NgbActiveModal} from '@ng-bootstrap/ng-bootstrap';
import {AppConfig} from 'jtb-core-games-ui';


let releaseNotes: string = 'We made some changes!';

class MockAppConfig implements AppConfig {
    appName: string = '';
    hoverMenu: boolean = false;
    releaseNotes: string = releaseNotes;
    version: string = '';
    inviteFriendsMessage: string = '';
}

class MockModal {
    dismiss = jasmine.createSpy('dismiss');
}

describe('Component:  default version notes component', () => {
    let fixture;
    let modal;
    beforeEach(() => {
        TestBed.configureTestingModule({
            declarations: [
                DefaultVersionNotesComponent,
            ],
            providers: [
                {provide: 'AppConfig', useClass: MockAppConfig},
                {provide: NgbActiveModal, useClass: MockModal}
            ],
        });
        TestBed.compileComponents();
        fixture = TestBed.createComponent(DefaultVersionNotesComponent);
        modal = TestBed.get(NgbActiveModal);
        fixture.detectChanges();
    });

    it('displays release notes', () => {
        expect(fixture.nativeElement.querySelector('.version-message').textContent.trim()).toEqual(releaseNotes);
    });

    it('close modal', () => {
        expect(modal.dismiss).not.toHaveBeenCalled();
        fixture.nativeElement.querySelector('.close-button').click();
        expect(modal.dismiss).toHaveBeenCalled();
    });
});
