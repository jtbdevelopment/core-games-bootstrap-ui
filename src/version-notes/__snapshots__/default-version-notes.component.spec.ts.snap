// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component:  default version notes component displays release notes 1`] = `
<ngbd-modal-content
  activeModal={[Function MockModal]}
  config={[Function MockAppConfig]}
>
  <div
    class="version-dialog"
  >
    
    
    <div
      class="modal-header"
    >
      
      
      <h3
        class="modal-title"
      >
        Welcome to version 
      </h3>
      
    
    </div>
    
    
    <div
      class="modal-body"
    >
      
      
      <span
        class="version-message"
      >
        We made some changes!
      </span>
      
    
    </div>
    
    
    <div
      class="modal-footer"
    >
      
      
      <button
        class="btn btn-info focus close-button"
      >
        OK
      
      </button>
      
    
    </div>
    
  
  </div>
  
</ngbd-modal-content>
`;
